### PR TITLE
Add cause of failures in ThriftHiveMetastore

### PR DESCRIPTION
**Pull request by pajaks**
This will be beneficial in case of detecting retry issues.
Currently in case of failure after retry (e.g createTable, dropTable) we have no information that it was consecutive attempt. After this  change exceptions from previous attempts will be visible in stack trace.

Example in case 2 attemps:
1. `ClientException`
2. `NoSuchObjectException`

Before:
```
io.trino.spi.connector.TableNotFoundException: Table 'test.test' not found

	at io.trino.plugin.hive.metastore.thrift.ThriftHiveMetastore.dropTable(ThriftHiveMetastore.java:1103)
	at io.trino.plugin.hive.TestTriftHiveMetastore.testDropTable(TestTriftHiveMetastore.java:194)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
```
After:
```
io.trino.spi.connector.TableNotFoundException: Table 'test.test' not found

	at io.trino.plugin.hive.metastore.thrift.ThriftHiveMetastore.dropTable(ThriftHiveMetastore.java:1103)
	at io.trino.plugin.hive.TestTriftHiveMetastore.testDropTable(TestTriftHiveMetastore.java:194)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
Caused by: NoSuchObjectException(message:null)
	at io.trino.plugin.hive.TestTriftHiveMetastore$2.getTable(TestTriftHiveMetastore.java:178)
	at io.trino.plugin.hive.metastore.thrift.ThriftHiveMetastore.lambda$dropTable$41(ThriftHiveMetastore.java:1092)
	at io.trino.plugin.hive.metastore.thrift.ThriftMetastoreApiStats.lambda$wrap$0(ThriftMetastoreApiStats.java:41)
	at io.trino.plugin.hive.util.RetryDriver.run(RetryDriver.java:116)
	at io.trino.plugin.hive.metastore.thrift.ThriftHiveMetastore.dropTable(ThriftHiveMetastore.java:1090)
	... 4 more
	Suppressed: io.trino.client.ClientException: Timout
		at io.trino.plugin.hive.TestTriftHiveMetastore$2.getTable(TestTriftHiveMetastore.java:176)
		... 8 more
```
'''

## Release notes

(x ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by findepi** [comment]
> This will be beneficial in case of detecting retry issues

Please expand rationale. Maybe example of the situation where current error reporting is insufficient and this change helps.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/TableAlreadyExistsException.java [hunk 1]
```diff
@@ -34,6 +34,11 @@ public TableAlreadyExistsException(SchemaTableName tableName, String message)
         this(tableName, message, null);
     }
 
+    public TableAlreadyExistsException(SchemaTableName tableName, Throwable cause)
```
Comment (findinpath) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/TableAlreadyExistsException.java:37]: You could make use of this constructor in 

https://github.com/trinodb/trino/blob/794ca2a8ce40d847054cbcec8cba2bb7789ecaa1/plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java#L566

**Comment by findinpath** [comment]
Failure in `ci / test (plugin/trino-delta-lake)`

```
Error:  io.trino.plugin.deltalake.TestDeltaLakeMinioAndHmsConnectorSmokeTest.testNonDeltaTablesCannotBeAccessed -- Time elapsed: 29.37 s <<< ERROR!
2023-11-08T03:23:48.900-0600	INFO	dispatcher-query-12	io.trino.event.QueryMonitor	TIMELINE: Query 20231108_092348_00631_xs34i :: FINISHED :: elapsed 144ms :: planning 3ms :: waiting 46ms :: scheduling 61ms :: running 59ms :: finishing 21ms :: begin 2023-11-08T03:23:48.755-06:00 :: end 2023-11-08T03:23:48.899-06:00
java.lang.IllegalStateException: result set must have exactly one row
	at com.google.common.base.Preconditions.checkState(Preconditions.java:512)
	at io.trino.testing.MaterializedResult.getOnlyValue(MaterializedResult.java:298)
	at io.trino.testing.AbstractTestQueryFramework.computeScalar(AbstractTestQueryFramework.java:323)
	at io.trino.testing.AbstractTestQueryFramework.computeScalar(AbstractTestQueryFramework.java:318)
	at io.trino.plugin.deltalake.BaseDeltaLakeConnectorSmokeTest.testNonDeltaTablesCannotBeAccessed(BaseDeltaLakeConnectorSmokeTest.java:495)
```

It is going to be likely fixed through #19343

**Review by findinpath — APPROVED** [review]

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/SchemaAlreadyExistsException.java [hunk 2]
```diff
@@ -25,12 +25,17 @@ public class SchemaAlreadyExistsException
 
     public SchemaAlreadyExistsException(String schemaName)
     {
-        this(schemaName, format("Schema already exists: '%s'", schemaName));
+        this(schemaName, format("Schema already exists: '%s'", schemaName), null);
```
Comment (ebyhr) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/SchemaAlreadyExistsException.java:28]: No need to declare the error message in two places:
```suggestion
        this(schemaName, null);
```

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/TableAlreadyExistsException.java (hunk 1 above)
Comment (ebyhr) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/TableAlreadyExistsException.java:37]: Can we remove the 2nd constructor (SchemaTableName tableName, String message)? 
```java
    public TableAlreadyExistsException(SchemaTableName tableName)
    {
        this(tableName, null);
    }

    public TableAlreadyExistsException(SchemaTableName tableName, Throwable cause)
    {
        this(tableName, format("Table already exists: '%s'", tableName), cause);
    }
```

**Review by ebyhr — APPROVED** [review]

**Pull request merged**